### PR TITLE
Fix Bluetooth setup_priorities

### LINKED ISSUE #2471: AirThings values are not coming through
### The problem

I have an AirThings Plus device which works fine through their mobile app BUT does  not work through BLE integration
though the device is discovered no issues, just values are not coming through the integration

### Which version of ESPHome has the issue?

 ESPHome version 2021.9.1

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

Home Assistant 2021.9.7

### What platform are you using?

ESP32

### Board

D1 esp32

### Component causing the issue

airthings_wave_plus

### Example YAML snippet

```yaml
sensor:
  - platform: airthings_wave_plus
    ble_client_id: den_airthings
    update_interval: 5min # default
    temperature:
      name: "WavePlus Temperature"
    radon:
      name: "WavePlus Radon"
    radon_long_term:
      name: "WavePlus Radon Long Term"
    pressure:
      name: "WavePlus Pressure"
    humidity:
      name: "WavePlus Humidity"
    co2:
      name: "WavePlus CO2"
    tvoc:
      name: "WavePlus VOC"

ble_client:
  - mac_address: "D8:71:4D:12:65:37"
    id: den_airthings

esp32_ble_tracker:
# airthings_ble:
```

### Anything in the logs that might be useful for us?

```txt
[21:16:58][C][ble_client:033]: BLE Client:
[21:16:58][C][ble_client:034]:   Address: d8:71:4d:12:65:37
[21:16:58][C][esp32_ble_tracker:614]: BLE Tracker:
[21:16:58][C][esp32_ble_tracker:615]:   Scan Duration: 300 s
[21:16:58][C][esp32_ble_tracker:616]:   Scan Interval: 320.0 ms
[21:16:58][C][esp32_ble_tracker:617]:   Scan Window: 30.0 ms
[21:16:58][C][esp32_ble_tracker:618]:   Scan Type: ACTIVE
[21:16:58][C][captive_portal:148]: Captive Portal:
[21:16:58][C][web_server:152]: Web Server:
[21:16:58][C][web_server:153]:   Address: den-bluetooth-bridge.local:80
[21:16:58][C][ota:029]: Over-The-Air Updates:
[21:16:58][C][ota:030]:   Address: den-bluetooth-bridge.local:3232
[21:16:58][C][ota:032]:   Using Password.
[21:16:58][C][api:135]: API Server:
[21:16:58][C][api:136]:   Address: den-bluetooth-bridge.local:6053
[21:16:58][C][wifi_signal.sensor:009]: WiFi Signal 'den-bluetooth-bridge - Wifi Signal'
[21:16:58][C][wifi_signal.sensor:009]:   Device Class: 'signal_strength'
[21:16:58][C][wifi_signal.sensor:009]:   State Class: 'measurement'
[21:16:58][C][wifi_signal.sensor:009]:   Unit of Measurement: 'dBm'
[21:16:58][C][wifi_signal.sensor:009]:   Accuracy Decimals: 0
[21:16:58][C][wifi_signal.sensor:009]:   Icon: 'mdi:wifi'
[21:17:05][D][sensor:121]: 'den-bluetooth-bridge - Wifi Signal': Sending state -28.00000 dBm with 0 decimals of accuracy
[21:17:34][D][sensor:121]: 'den-bluetooth-bridge - Uptime': Sending state 90.51700 s with 0 decimals of accuracy
[21:18:05][D][sensor:121]: 'den-bluetooth-bridge - Wifi Signal': Sending state -28.00000 dBm with 0 decimals of accuracy
[21:18:34][D][sensor:121]: 'den-bluetooth-bridge - Uptime': Sending state 150.51199 s with 0 decimals of accuracy
[21:18:58][W][airthings_wave_plus:106]: Connection in progress
[21:19:05][D][sensor:121]: 'den-bluetooth-bridge - Wifi Signal': Sending state -29.00000 dBm with 0 decimals of accuracy
[21:19:34][D][sensor:121]: 'den-bluetooth-bridge - Uptime': Sending state 210.51199 s with 0 decimals of accuracy
[21:20:05][D][sensor:121]: 'den-bluetooth-bridge - Wifi Signal': Sending state -29.00000 dBm with 0 decimals of accuracy
[21:20:34][D][sensor:121]: 'den-bluetooth-bridge - Uptime': Sending state 270.51199 s with 0 decimals of accuracy
[21:21:04][I][ota:046]: Boot seems successful, resetting boot loop counter.
[21:21:05][D][sensor:121]: 'den-bluetooth-bridge - Wifi Signal': Sending state -29.00000 dBm with 0 decimals of accuracy
[21:21:34][D][sensor:121]: 'den-bluetooth-bridge - Uptime': Sending state 330.51199 s with 0 decimals of accuracy
[21:22:05][D][sensor:121]: 'den-bluetooth-bridge - Wifi Signal': Sending state -28.00000 dBm with 0 decimals of accuracy
[21:22:34][D][sensor:121]: 'den-bluetooth-bridge - Uptime': Sending state 390.51599 s with 0 decimals of accuracy
[21:23:05][D][sensor:121]: 'den-bluetooth-bridge - Wifi Signal': Sending state -29.00000 dBm with 0 decimals of accuracy
[21:23:34][D][sensor:121]: 'den-bluetooth-bridge - Uptime': Sending state 450.51599 s with 0 decimals of accuracy
[21:23:58][W][airthings_wave_plus:106]: Connection in progress
[21:24:05][D][sensor:121]: 'den-bluetooth-bridge - Wifi Signal': Sending state -30.00000 dBm with 0 decimals of accuracy
[21:24:34][D][sensor:121]: 'den-bluetooth-bridge - Uptime': Sending state 510.51700 s with 0 decimals of accuracy
[21:25:05][D][sensor:121]: 'den-bluetooth-bridge - Wifi Signal': Sending state -29.00000 dBm with 0 decimals of accuracy
[21:25:34][D][sensor:121]: 'den-bluetooth-bridge - Uptime': Sending state 570.51202 s with 0 decimals of accuracy
[21:26:05][W][esp32_ble_tracker:196]: ESP-IDF BLE scan never terminated, rebooting to restore BLE stack...
[21:26:05][I][app:127]: Forcing a reboot...
INFO Disconnected from ESPHome API for den-bluetooth-bridge.local
WARNING Disconnected from API
INFO den-bluetooth-bridge.local: Unexpected error while reading incoming messages: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/aioesphomeapi/connection.py", line 505, in run_forever
    await self._run_once()
  File "/usr/local/lib/python3.9/dist-packages/aioesphomeapi/connection.py", line 479, in _run_once
    assert self._frame_helper is not None
AssertionError
INFO Successfully connected to den-bluetooth-bridge.local
[21:26:52][D][sensor:121]: 'den-bluetooth-bridge - Uptime': Sending state 46.74900 s with 0 decimals of accuracy
[21:27:02][D][sensor:121]: 'den-bluetooth-bridge - Wifi Signal': Sending state -30.00000 dBm with 0 decimals of accuracy
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jesserockz**
# What does this implement/fix? 

Set ble_tracker codegen priority so it gets added before clients

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes esphome/issues#2471

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [x] ESP32
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Review by OttoWinter — COMMENTED** [review]
LGTM, but:

Don't clients already implicitly wait for the tracker for the setup? `register_ble_device` calls `get_variable` which only returns after the ESP32BLETracker variable has been declared by esp32_ble_tracker->to_code

**Comment by jesserockz** [comment]
Ah the issue is not the order of the code in main.cpp, but actually the setup_priority (masked by the order added to main.cpp).

Setup is being run for the ble_client before the tracker if the tracker was lower in yaml

File: esphome/components/ble_client/ble_client.cpp [hunk 1]
```diff
@@ -11,6 +11,8 @@ namespace ble_client {
 
 static const char *const TAG = "ble_client";
 
+float ESP32BLETracker::get_setup_priority() const { return setup_priority::AFTER_BLUETOOTH; }
```
Comment (OttoWinter) [inline esphome/components/ble_client/ble_client.cpp]:
```suggestion
float BLEClient::get_setup_priority() const { return setup_priority::AFTER_BLUETOOTH; }
```

Comment (jesserockz) [reply esphome/components/ble_client/ble_client.cpp]: Woops, cheers

**Pull request merged**